### PR TITLE
SSO implementation - Support for use of apimlAuthenticationToken

### PR DESCRIPTION
**Pull request by jordanCain**
Change Authentication so that now all requests must include an apimlAuthenticationToken either as a cookie header called apimlAuthenticationToken or in the Authroization header. 

Requests to z/OSMF are now routed via the gateway so that the LTPA can be extracted from the apiml JWT.

### CODE REVIEW

**Comment by jordanCain** [comment]
Sonar Cloud code smells are down to following existing project style, 

One issue has been raised here to fix the sonarcloud rule https://github.com/zowe/explorer-api-common/issues/42

File: explorer-api-common-test/src/main/java/org/zowe/api/common/test/AbstractHttpIntegrationTest.java [hunk 1]
```diff
@@ -13,11 +13,13 @@
 import io.restassured.http.ContentType;
```
Comment (stevenhorsman) [inline explorer-api-common-test/src/main/java/org/zowe/api/common/test/AbstractHttpIntegrationTest.java]: Copyright update

File: explorer-api-common/src/main/java/org/zowe/api/common/security/SpringSecurityConfig.java [hunk 2]
```diff
@@ -5,44 +5,27 @@
...
     private static final String[] AUTH_WHITELIST = {
             // -- swagger ui
             "/swagger-resources/**", "/swagger-ui.html", "/v2/api-docs", "/webjars/**"};
 
     @Override
     protected void configure(HttpSecurity http) throws Exception {
         // TODO - re-enable csrf?
-        http.csrf().disable().authorizeRequests().antMatchers(AUTH_WHITELIST).permitAll().antMatchers("/api/**/*")
-                .authenticated().and().httpBasic().authenticationEntryPoint(authEntryPoint).and()
-                .logout().logoutRequestMatcher(new AntPathRequestMatcher("/api/v1/**/logout"));
```
Comment (stevenhorsman) [inline explorer-api-common/src/main/java/org/zowe/api/common/security/SpringSecurityConfig.java]: I guess that the logout mechanism is by just logging out from the gateway now?

Comment (jordanCain) [reply explorer-api-common/src/main/java/org/zowe/api/common/security/SpringSecurityConfig.java]: APIML squad is working on a proper logout endpoint that will invalidate the JWT, which will be the logout mechanism. Once that is delivered we should also front that with an api in each of data-sets and jobs for completeness. but a user could just use the endpoint directly with the gateway

**Pull request merged**